scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1654,126 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if type(shape) != tuple:
+        # in case shape is not passed in as a tuple
+        shape = (shape,)
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_pttrf_pttrs(dtype):
+    np.random.seed(42)
+    # set test tolerance appropriate for dtype
+    rtol = 250*np.finfo(dtype).eps
+    atol = 100*np.finfo(dtype).eps
+    # n is the length diagonal of A
+    n = 10
+    # create diagonals according to size and dtype
+    if dtype in REAL_DTYPES:
+        # add 2 so that the matrix will be diagonally dominant
+        d = generate_random_dtype_array(n, dtype) + 2
+    else:
+        # diagonal d should always be real.
+        # for complex add 4 so it will be dominant
+        d = generate_random_dtype_array(n, DTYPES[1]) + 4
```

Comment:
```suggestion
        d = generate_random_dtype_array(n, np.float64) + 4
```

Reply:
Actually, although you only want real values, you should make sure `dtype` is respected. So for `complex64`, this should be of type `complex64` with all imaginary parts 0. This might be why the test is not passing.

Reply:
Interesting- right now it looks like the wrapper converts to real and raises a complex discard warning. I assume that passing in a complex type vs a real type array won't make any difference right now while the imaginary parts get stripped anyways. 

Changing this (with the current wrapper) doesn't appear to have an effect on the outputs, they still fail the tests.

Reply:
Let's take a look together this week?

Reply:
Sounds good! Office hours tomorrow are good for me. I can also come Wednesday morning.

Reply:
My comment above was wrong. We found the real mistake in office hours.